vllm/model_executor/layers/attention.py

```diff
@@ -101,23 +101,15 @@ def forward(
         # vectors will not be cached. This happens during the initial memory
         # profiling run.
         if key_cache is not None and value_cache is not None:
-            key_to_cache = key
-            value_to_cache = value
-            if input_metadata.to_cache is not None:
-                key_to_cache = key_to_cache[input_metadata.to_cache]
-                value_to_cache = value_to_cache[input_metadata.to_cache]
-                slot_mapping = slot_mapping[input_metadata.to_cache]
```

Comment:
In this case, how does sliding window caching work?

Reply:
That's actually a very good question. The `reshape_and_cache` kernel does nothing if the corresponding slot ID of the token is -1 (`_PAD_SLOT_ID`). Leveraging this, the new logic in `_prepare_prompt` mask the `[0, start_idx)` tokens of the prompt, where `start_idx` is `max(0, prompt_len - sliding_window`). In other words, if the prompt len is 10, sliding window is 8, and block size is 4, the slot mapping will be `[-1, -1, 2, 3, 4, 5, 6, 7, 0, 1]`.

Reply:
got it! Can we leave this as a comment somewhere in the code? Maybe in `PagedAttention` class?

Reply:
Yes I will, but I believe `PagedAttention` class is not a good place, since the logic is now removed. I will add this into the `_prepare_prompt` method.